torch/optim/lr_scheduler.py

```diff
@@ -82,12 +121,23 @@ def step(self, epoch=None):
                               "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
         self._step_count += 1
 
+
         if epoch is None:
-            epoch = self.last_epoch + 1
-        self.last_epoch = epoch
-        for param_group, lr in zip(self.optimizer.param_groups, self.get_lr()):
+            self.last_epoch = self.last_epoch + 1
+            values = self._compute_values()
+        else:
+            warnings.warn(EPOCH_DEPRECATION_WARNING, DeprecationWarning)
```

Comment:
is `DeprecationWarning` ignored by default? https://github.com/pytorch/pytorch/blob/92750acb88ec5539658b1c51ee27a648ce92a33a/torch/nn/modules/container.py#L15-L17

Reply:
I see the deprecation warning being raised when running the `test/test_optim.py` without warning filters modified. Do you know the context of that warning?